### PR TITLE
Added i3 and its dependencies.

### PR DESCRIPTION
**Pull request by stefanosoffia**
Added package for [i3 window manager](i3wm.org) including the needed dependencies.

### CODE REVIEW

File: var/spack/repos/builtin/packages/flex/package.py [hunk 1]
```diff
@@ -32,7 +32,9 @@ class Flex(AutotoolsPackage):
     homepage = "https://github.com/westes/flex"
     url = "https://github.com/westes/flex/releases/download/v2.6.1/flex-2.6.1.tar.gz"
 
-    version('2.6.4', '2882e3179748cc9f9c23ec593d6adc8d')
+    # Avoid flex '2.6.4' because it fails compilation with GCC 7.2.0 Linux x64.
+    # See: https://github.com/westes/flex/issues/219
+    # version('2.6.4', '2882e3179748cc9f9c23ec593d6adc8d')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flex/package.py]: I would make this change in another PR. I don't think we should remove this version altogether. Let's decide on something in #6812 and leave it for another PR.

**Comment by stefanosoffia** [comment]
Your request is very reasonable. I cleaned up this PR and moved the flex related changes to PR #6819.

File: var/spack/repos/builtin/packages/libev/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class Libev(AutotoolsPackage):
+    """A full-featured and high-performance event loop that is loosely modelled
+    after libevent, but without its limitations and bugs."""
+
+    homepage = "http://software.schmorp.de/pkg/libev.html"
+    url      = "https://github.com/enki/libev"
+
+    version('develop', git='https://github.com/enki/libev',
+            commit='93823e6ca699df195a6c7b8bfa6006ec40ee0003')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libev/package.py]: One last request. Can you add a stable release as well? We like to see at least one stable release in every package. If you add:
```python
url      = "http://dist.schmorp.de/libev/libev-4.24.tar.gz"
list_url = "http://dist.schmorp.de/libev/Attic/"

version('4.24', '94459a5a22db041dec6f98424d6efe54')
```
the `list_url` will allow `spack versions` and `spack checksum` to pick up every version.

File: var/spack/repos/builtin/packages/libev/package.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libev/package.py]: These dependencies may only be needed `when='@develop'`.

File: var/spack/repos/builtin/packages/xcb-util-xrm/package.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    homepage = "http://www.example.com"
+    url      = "https://github.com/Airblader/xcb-util-xrm/archive/v1.2.tar.gz"
+
+    # This GitHub project includes some git submodules, which must be fetched
+    # in order to build it.
+    version('1.2',
+            git='https://github.com/Airblader/xcb-util-xrm',
+            tag='v1.2',
+            submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xcb-util-xrm/package.py]: Any reason to use the git tag instead of the archive tarball? It's easier to checksum a tarball, which is nice for security reasons. The tarball often doesn't require any of the Autotools dependencies either.

File: var/spack/repos/builtin/packages/xcb-util-xrm/package.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class XcbUtilXrm(AutotoolsPackage):
+    """XCB util-xrm module provides the 'xrm' library, i.e.  utility functions
+    for the X resource manager."""
+
+    homepage = "http://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xcb-util-xrm/package.py]: Just noticed this, can you update the homepage to https://github.com/Airblader/xcb-util-xrm ?

File: var/spack/repos/builtin/packages/xcb-util-xrm/package.py (hunk 4 above)
Comment (stefanosoffia) [reply var/spack/repos/builtin/packages/xcb-util-xrm/package.py]: Thanks for your patience, I addressed your points but the last one. Specifically, according to [this conversation](https://github.com/Airblader/xcb-util-xrm/issues/77), `xcb-util-xrm` project has git submodules which are not included into the release archive. My understanding is that Spack `version` argument `submodules` is only considered when the sources are retrieved through a git repo.

File: var/spack/repos/builtin/packages/libev/package.py (hunk 2 above)
Comment (stefanosoffia) [reply var/spack/repos/builtin/packages/libev/package.py]: Fixed.

File: var/spack/repos/builtin/packages/libev/package.py (hunk 3 above)
Comment (stefanosoffia) [reply var/spack/repos/builtin/packages/libev/package.py]: Fixed by adding `when='@develop'`.
